src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -138,6 +144,8 @@ public AddressBook load() throws StorageOperationException {
             throw new StorageOperationException("Error parsing file data format");
         } catch (IllegalValueException ive) {
             throw new StorageOperationException("File contains illegal data values; data type constraints not met");
+        } catch (RuntimeException rte) {
+        	throw new FileDeletedWhenRunningException("File deleted when program is running");
```

Comment:
- Glad you attempted this and introduced the new exception class too. But your way of handling the exception is not good. Take note of the following.

- First where you are throwing the exception is not appropriate. save() is a better places to see if the file exists and recover and throw the exception

- You have not implemented any code to handle the exception you are throwing. hint : show a warning to the user in  Main.java